src/clib/lib/job_queue/torque_driver.cpp

```diff
@@ -615,13 +615,17 @@ torque_driver_get_qstat_status(torque_driver_type *driver,
            retry a couple of times with exponential sleep. ERT pings qstat
            every second for every realization, thus the initial sleep time
            is 2 seconds. */
+        int return_value = -1;
         bool qstat_succeeded = false;
         int retry_interval = 2; /* seconds */
         int slept_time = 0;
         while ((!qstat_succeeded) && (slept_time <= driver->timeout)) {
-            util_spawn_blocking(driver->qstat_cmd, argv.size(), argv.data(),
-                                tmp_std_file, tmp_err_file);
-            if (fs::exists(tmp_std_file) && (fs::file_size(tmp_std_file) > 0)) {
+            return_value =
+                util_spawn_blocking(driver->qstat_cmd, argv.size(), argv.data(),
+                                    tmp_std_file, tmp_err_file);
+            // A zero return-value is not trusted, but a nonzero is.
+            if (fs::exists(tmp_std_file) && (fs::file_size(tmp_std_file) > 0) &&
+                return_value == 0) {
                 // A non-existing or zero length output from qstat is interpreted
```

Comment:
I guess this comment does not make sense then or?

Reply:
Yes, it does make sense. But I can see it can be hard to make sense of it.

Reply:
Comments rewritten.